### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by Vsevolod-Livinskij**
It is very preliminary and doesn't support a lot of features.

### CODE REVIEW

File: check_isa.cpp [hunk 1]
```diff
@@ -32,20 +38,33 @@ static void __cpuidex(int feature_regs[4], int level, int count) {
                           : "=a" (feature_regs[0]), "=r" (feature_regs[1]), "=c" (feature_regs[2]), "=d" (feature_regs[3])
                           : "0" (level), "2" (count));
 }
+#endif // !WINDOWS
 
 static int call_xgetbv() {
+#ifndef WINDOWS
     // Call xgetbv. We use byte sequence in order to support older assemblers.
     int eax, edx;
-    __asm__ __volatile__ (".byte 0x0f, 0x01, 0xd0" : "=a" (eax), "=d" (edx) : "c" (0));
```
Comment (dbabokin) [inline check_isa.cpp:39]: What about just borrowing check_isa.cpp from ISPC project?
https://github.com/ispc/ispc/blob/master/check_isa.cpp

It was originally taken from there.

File: src/main.cpp [hunk 2]
```diff
@@ -68,6 +68,7 @@ void print_usage_and_exit (std::string error_msg = "") {
     std::cout << "\t-d, --out-dir=<out-dir>   Output directory\n";
     std::cout << "\t-s, --seed=<seed>         Predefined seed (it is accepted in form of SSS or VV_SSS)\n";
     std::cout << "\t-m, --bit-mode=<32/64>    Generated test's bit mode\n";
+    std::cout << "\t--windows=<true/false>    Enable windows-compatible mode\n";
```
Comment (dbabokin) [inline src/main.cpp:71]: Why adding it as a command line option, I don't think that we care about cross compilation. And we can detect Windows host when compiling on Windows, right?

File: check_isa.cpp (hunk 1 above)
Comment (Vsevolod-Livinskij) [reply check_isa.cpp:39]: Fixed

File: src/main.cpp (hunk 2 above)
Comment (Vsevolod-Livinskij) [reply src/main.cpp]: Fixed

**Pull request merged**